### PR TITLE
Show docs and other stuff.

### PR DESCRIPTION
**Pull request by Paul-Ferrell**

### CODE REVIEW

**Comment by Paul-Ferrell** [comment]
Fixes #285

**Comment by Paul-Ferrell** [comment]
Fixes #284

**Comment by Paul-Ferrell** [comment]
Fixes #87

**Comment by Paul-Ferrell** [comment]
Fixes #215

**Comment by Paul-Ferrell** [comment]
closes #215 
closes #87 
closes #284 
closes #285

File: lib/pavilion/builder.py [hunk 1]
```diff
@@ -230,6 +232,21 @@ def exists(self):
...
 
+    def log_updated(self) -> Union[float, None]:
+        """Return the last time the build log was updated."""
+
+        if self.tmp_log_path.exists():
+            try:
+                return self.tmp_log_path.stat().st_mtime
+            except OSError:
+                pass
+
```
Comment (j-ogas) [inline lib/pavilion/builder.py:243]: Under what circumstance might we trigger an OS error here? Why do we pass if we encounter one?

File: lib/pavilion/plugins/commands/show.py [hunk 2]
```diff
@@ -776,6 +783,52 @@ def _tests_cmd(self, pav_cfg, args):
...
+        """Show the documentation for the requested test."""
+
+        resolv = resolver.TestConfigResolver(pav_cfg)
+        suites = resolv.find_all_tests()
+
+        parts = args.test_name.split('.')
+        if len(parts) != 2:
+            output.fprint(
+                "You must give a test name as '<suite>.<test>'.",
+                file=self.outfile, color=output.RED)
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply lib/pavilion/plugins/commands/show.py:796]: I think it's more important to know that your format is broken.

File: lib/pavilion/result/__init__.py [hunk 3]
```diff
@@ -4,11 +4,37 @@
...
+    def log(msg, lvl=None):
+        """Log the given message to the log_file."""
+
+        nonlocal log_tab_level
+
+        if lvl is not None:
+            log_tab_level = lvl
+
+        if log_file is not None:
+            log_file.write(textwrap.indent(msg, "  " * log_tab_level))
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply lib/pavilion/result/__init__.py:34]: IO[str] is an open file, so yeah, it has to handed an open file.

File: lib/pavilion/plugins/commands/show.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicholas-sly) [reply lib/pavilion/plugins/commands/show.py:796]: Definitely specify that the format isn't correct.  But also giving the appropriate things to add would be helpful in speeding up the turnaround of this failure.

**Pull request merged**